lib/gringotts/gateways/pin_pay.ex

```diff
@@ -0,0 +1,423 @@
+  > that true for PinPay?
+  > Is there a limited time window within which a void can be perfomed?
+
+  ## Example
+
+  > A barebones example using the bindings you've suggested in the `moduledoc`.
+  """
+  @spec refund(Money.t, String.t(), keyword) :: {:ok | :error, Response}
+  def refund(amount, payment_id, opts) do
+    url=@test_url <> "charges/" <> payment_id <> "/refunds"
+    commit(:post, url)
+  end
+
+
+  @doc """
+  Stores the payment-source data for later use.
+
+  > This usually enable "One Click" and/or "Recurring Payments"
+
+  ## Note
+
+  > If there's anything noteworthy about this operation, it comes here.
+
+  ## Example
+
+  > A barebones example using the bindings you've suggested in the `moduledoc`.
+  """
+  
+  @spec store(CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def store(%CreditCard{} = card, opts) do
+    commit(:post, "cards", card_for_token(card, opts), opts)
+  end
+
+  @doc """
+  Removes card or payment info that was previously `store/2`d
+
+  Deletes previously stored payment-source data.
+
+  ## Note
+
+  > If there's anything noteworthy about this operation, it comes here.
+
+  ## Example
+
+  > A barebones example using the bindings you've suggested in the `moduledoc`.
+  """
+  @spec unstore(String.t(), keyword) :: {:ok | :error, Response}
+  def unstore(registration_id, opts) do
+    # can't be implemented in pinpayments
+  end
+
+  ###############################################################################
+  #                                PRIVATE METHODS                              #
+  ###############################################################################
+
+  # Makes the request to PinPay's network.
+  # For consistency with other gateway implementations, make your (final)
+  # network request in here, and parse it using another private method called
+  # `respond`.
+
+  defp card_params(card, opts) do
+    [
+      "card[number]": card.number,
+      "card[name]": card.first_name <> card.last_name,
+      "card[expiry_month]": card.month |> Integer.to_string() |> String.pad_leading(2, "0"),
+      "card[expiry_year]": card.year |> Integer.to_string(),
+      "card[cvc]": card.verification_code,
+      "card[address_line1]": opts[:address].street1,
+      "card[address_city]": opts[:address].city,
+      "card[address_country]": opts[:address].country
+    ]
+  end
+
+  defp card_for_token(card, opts) do
+    [
+      "number": card.number,
+      "name": card.first_name <> card.last_name,
+      "expiry_month": card.month |> Integer.to_string() |> String.pad_leading(2, "0"),
+      "expiry_year": card.year |> Integer.to_string(),
+      "cvc": card.verification_code,
+      "address_line1": opts[:Address][:street1],
+      "address_city": opts[:Address][:city],
+      "address_country": opts[:Address][:country]
+    ]
+  end
+
+  @spec commit(atom, String.t(), keyword, keyword) :: {:ok | :error, Response}
+  defp commit(:post, endpoint, param, opts) do
```

Comment:
The `opts` argument is not being used in this function, but on line 176, 189 you've passed an important field: `currency`.